www/src/js/views/snap-details.js

```diff
@@ -0,0 +1,254 @@
+      <SnapChannel {...props.model.attributes} />
+      <SnapInstallDate {...props.model.attributes} />
+    </ul>
+  );
+}
+
+function About(props) {
+  var model = props.model
+  if (model.get('description')) {
+    return (
+      <div>
+        <h3>About</h3>
+        <p>{model.get('description')}</p>
+      </div>
+    );
+  }
+  return (
+    <div>
+      <h3>About</h3>
+      <p><i>No description</i></p>
+    </div>
+  );
+}
+
+function SnapBanner(props) {
+  var model = props.model;
+  return (
+    <div>
+      <div className="col-2">
+        <div className="b-icon-wrapper">
+          <img className="b-icon" src={model.get('icon')} width="125" height="125" />
+        </div>
+      </div>
+      <div className="col-6">
+        <h1 className="b-snap__title">{model.get('name')}</h1>
+        <div className="b-snap__developer">{model.get('developer')}</div>
+      </div>
+    </div>
+  );
+}
+
+function SnapInterfaces(props) {
+  return (
+    <ul className="p-list--divided">
+      <li className="p-list__item">Interfaces</li>
+      <li className="p-list__item" id="interface-list">N/A</li>
+    </ul>
+  );
+}
+
+function SnapActions(props) {
+  // download_progress
+  var model = props.model;
+  var progressWidth = props.downloadProgress + "%";
+  var status = props.status;
+  var taskSummary = props.taskSummary;
+
+  var progressBarBorder = "1px";
+  var progressBarBorderRadius = "10%";
+  var progressBarBackgroundColor = "#003399";
+
+  var installerClass = "b-installer_small";
+
+  if (status !== CONF.INSTALL_STATE.INSTALLING) {
+    progressBarBorder = "0px";
+    progressBarBorderRadius = "initial";
+    progressBarBackgroundColor = "";
+  }
+
+  if (status === CONF.INSTALL_STATE.INSTALLING ||
+      status === CONF.INSTALL_STATE.REMOVING) {
+    installerClass += " b-installer_thinking";
+  }
+
+  var installerButtonClassName = "col-5";
+  if (status === CONF.INSTALL_STATE.INSTALLED ||
+      status === CONF.INSTALL_STATE.ACTIVE) {
+    installerButtonClassName = "col-2";
+  }
+  if (status === CONF.INSTALL_STATE.REMOVED) {
+    installerButtonClassName = "col-5";
+  }
+
+  return (
+    <div id="installer-button" className={installerButtonClassName}>
+      <Installer
+         installerClass={installerClass}
+         model={model} />
+
+      <div id="progressbarwrapper"
+           style={{border: {progressBarBorder}, borderRadius: {progressBarBorderRadius}, marginTop: "1em", width: "100%", height: "0.5em"}}>
+        <div id="progress"
+             style={{backgroundColor: {progressBarBackgroundColor}, width: {progressWidth}, height: "0.5em"}} />
+      </div>
+
+      <div style={{width: "100%", height: "23px", whiteSpace: "nowrap", textOverflow: "ellipsis", overflow: "hidden"}}>
+        <small id="tasksummary">{taskSummary}</small>
+      </div>
+    </div>
+  );
+}
+
+module.exports = React.createBackboneClass({
+  componentWillMount: function() {
+    var model = this.props.model;
+    if (! model) {
```

Comment:
well ... even if trapped above, ... I tend to be very paranoid in js ... there are not real guarantees at the component level that we can put to force a non null model ... which makes me uneasy about removing it?